lib/src/rules/conformance/conformance_rule.dart

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../../analyzer.dart';
+
+/// Base type for all compliance checks for web libraries.
+class ConformanceRule extends LintRule {
```

Comment:
This class doesn't appear to add much value at the moment. Do you have a vision for additional state/behavior that might be added in the future? If not, consider removing it until we know there's a need for it.

Reply:
Yes. I'll likely expect to create classes that place bans on property gets, non-constant property writes, methods/constructors, and particular imports, which is why I have a base class here for the different types of conformance rules. I might also add another subclass here to distinguish between conformance rules that involve bindings (like property gets) and ones that don't (like imports).

Reply:
Ok. I'd tend to wait to create a base class until I have demonstrated a need for it, but that's just my working style and not something you should feel compelled to copy.

Reply:
Yeah, I can see that being cleaner. I could remove it in this CL and adding it back in a future one, but that seems like a double review of this class. :P